examples/simple_reverse.coffee

```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env coffee
+# This is a simple example of how to use the slack-client module in CoffeeScript. It creates a
+# bot that responds to all messages in all channels it is in with a reversed
+# string of the text received.
+#
+# To run, copy your token below, then:
+#  npm install
+#   cd examples
+#   coffee simple_reverse.coffeescript
+
+Slack = require '..'
+
+token = 'xoxb-YOUR-TOKEN-HERE' # Add a bot at https://my.slack.com/services/new/bot and copy the token here.
+autoReconnect = true
+autoMark = true
+
+slack = new Slack(token, autoReconnect, autoMark)
+
+slack.on 'open', ->
+  channels = []
+  groups = []
+  unreads = slack.getUnreadCount()
+  key
+
+  # Get all the channels that bot is a member of
+  channels = ("##{value.name}" for key, value of slack.channels when slack.channels[key].is_member)
+  # Get all groups that are open and not archived 
+  groups = (value.name for key, value of slack.groups when slack.groups[key].is_open and not slack.groups[key].is_archived)
```

Comment:
This is probably getting on the verbose side for what I'd consider for a one liner.  It is correct, just might be a little more readable if it was split up.